### PR TITLE
Cache ElementwiseKernel object

### PR DESCRIPTION
**Pull request by okuta**
This PR avoids recreate ElementwiseKernel object.

### CODE REVIEW

**Comment by okuta** [comment]
Jenkins, test this please

**Comment by okuta** [comment]
Jenkins, test this please

File: cupyx/scipy/ndimage/filters.py [hunk 1]
```diff
@@ -106,12 +107,8 @@ def _correlate_or_convolve(input, weights, output, mode, cval, origin,
         return output
     input = cupy.ascontiguousarray(input)
     weights = cupy.ascontiguousarray(weights, cupy.float64)
-    # weights is always casted to float64 in order to get an output compatible
-    # with SciPy, thought float32 might be sufficient when input dtype is low
-    # precision.
-    in_params, out_params, operation, name = _generate_correlete_kernel(
-        input.ndim, mode, cval, input.shape, wshape, origin)
-    return cupy.ElementwiseKernel(in_params, out_params, operation, name)(
+    return _get_correlete_kernel(
```
Comment (emcastillo) [inline cupyx/scipy/ndimage/filters.py]: typo here `_get_correlate_kernel`

File: cupyx/scipy/ndimage/filters.py [hunk 2]
```diff
@@ -198,3 +195,13 @@ def _generate_correlete_kernel(ndim, mode, cval, xshape, wshape, origin):
         ndim, mode, '_'.join(['{}'.format(j) for j in xshape]),
         '_'.join(['{}'.format(j) for j in wshape]))
     return in_params, out_params, operation, name
+
+
+@util.memoize()
+def _get_correlete_kernel(ndim, mode, cval, xshape, wshape, origin):
```
Comment (emcastillo) [inline cupyx/scipy/ndimage/filters.py]: same `_get_correlate_kernel`

File: cupyx/scipy/ndimage/filters.py (hunk 1 above)
Comment (emcastillo) [inline cupyx/scipy/ndimage/filters.py]: There is a problem when memoizing a list here, this is the source of Jenkins errors

**Comment by okuta** [comment]
Thanks, I fixed.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by emcastillo** [comment]
Same error 😞
```
  File "/work/cupy/tests/cupyx_tests/scipy_tests/ndimage_tests/test_filters.py", line 72, in test_convolve_and_correlate
E       return self._filter(xp, scp, a, w)
E     File "/work/cupy/tests/cupyx_tests/scipy_tests/ndimage_tests/test_filters.py", line 60, in _filter
E       cval=self.cval, origin=origin)
E     File "/work/cupy/cupyx/scipy/ndimage/filters.py", line 64, in convolve
E       True)
E     File "/work/cupy/cupyx/scipy/ndimage/filters.py", line 111, in _correlate_or_convolve
E       input.ndim, mode, cval, input.shape, tuple(wshape), origin)(
E     File "cupy/util.pyx", line 79, in cupy.util.memoize.decorator.ret
E       result = m.get(arg_key, m)
E   TypeError: unhashable type: 'list'
```

**Comment by okuta** [comment]
jenkins, test this please.

**Pull request merged**